### ISSUE TITLE
Private gems in a configuration block?

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-cli#110]
From @abunashir:

> I've added a temporary fix, for now, it will silently ignore loading issue for this private gems. in the future, we also consider having those as configuration block.

_Originally posted by @abunashir in https://github.com/metanorma/metanorma-cli/issues/104#issuecomment-517925598_

What did you mean by "configuration block"? I think we need a good way of loading private gems. Right now, we load them by inferring on the document flavor in the command line / file attribute:

e.g.

* Command line `metanorma -t {private-type} document.adoc`

* In document: `:mn-document-type: {private-type}`

Then metanorma-cli will automatically load the gem `metanorma-{private-type}`.

**Comment by abunashir** [comment]
I was thinking to have some interface that will allow a user to add their custom backend, more likely `metanorma backends add [backend name]`. That way we can only focus on the public backends and if anyone wants to add their own backend / private backend then they can do so.

**Comment by ronaldtse** [comment]
Good idea. This goes back to our previous discussion of adding template repositories, needing a global metanorma config. This is also necessary for document repositories. 

However the word “backend” may be a bit confusing. “Processor” sounds better. 

That said “metanorma -t my-private-type blah.adoc” works equally well.
